src/exception/call_stack/elf.cr

```diff
@@ -0,0 +1,73 @@
+require "crystal/elf"
+
+struct Exception::CallStack
+  protected def self.load_dwarf
+    phdr_callback = LibC::DlPhdrCallback.new do |info, size, data|
+      # HACK: this assumes the first entry is the header for the current program
```

Comment:
Seems like `dl_iterate_phdr` is guaranteed to iterate in the order which ELF objects are loaded, so unless crystal is being used in a shared library, this would be correct.

Reply:
Yes, that’s true, at least it’s in glibc and musl. We should check if that assumption works on BSD too. At first I was using the name, which in glibc is empty for the main program. But musl returns /proc/self/exe.